mm2src/coins/tendermint/tendermint_token.rs

```diff
@@ -98,134 +100,64 @@ impl SwapOps for TendermintToken {
             .send_taker_fee_for_denom(fee_addr, amount, self.denom.clone(), self.decimals, uuid)
     }
 
-    fn send_maker_payment(
-        &self,
-        time_lock_duration: u64,
-        _time_lock: u32,
-        taker_pub: &[u8],
-        secret_hash: &[u8],
-        amount: BigDecimal,
-        _swap_contract_address: &Option<BytesJson>,
-        _swap_unique_data: &[u8],
-        _payment_instructions: &Option<PaymentInstructions>,
-    ) -> TransactionFut {
+    fn send_maker_payment(&self, maker_payment_args: SendMakerPaymentArgs) -> TransactionFut {
         self.platform_coin.send_htlc_for_denom(
-            time_lock_duration,
-            taker_pub,
-            secret_hash,
-            amount,
+            maker_payment_args.time_lock_duration,
+            maker_payment_args.pubkey,
+            maker_payment_args.secret_hash,
+            maker_payment_args.amount,
             self.denom.clone(),
             self.decimals,
         )
     }
 
-    fn send_taker_payment(
-        &self,
-        time_lock_duration: u64,
-        _time_lock: u32,
-        maker_pub: &[u8],
-        secret_hash: &[u8],
-        amount: BigDecimal,
-        swap_contract_address: &Option<BytesJson>,
-        swap_unique_data: &[u8],
-        _payment_instructions: &Option<PaymentInstructions>,
-    ) -> TransactionFut {
+    fn send_taker_payment(&self, taker_payment_args: SendTakerPaymentArgs) -> TransactionFut {
         self.platform_coin.send_htlc_for_denom(
-            time_lock_duration,
-            maker_pub,
-            secret_hash,
-            amount,
+            taker_payment_args.time_lock_duration,
+            taker_payment_args.pubkey,
+            taker_payment_args.secret_hash,
+            taker_payment_args.amount,
             self.denom.clone(),
             self.decimals,
         )
     }
 
     fn send_maker_spends_taker_payment(
         &self,
-        taker_payment_tx: &[u8],
-        time_lock: u32,
-        taker_pub: &[u8],
-        secret: &[u8],
-        secret_hash: &[u8],
-        swap_contract_address: &Option<BytesJson>,
-        swap_unique_data: &[u8],
+        maker_spends_payment_args: SendMakerSpendsTakerPaymentArgs,
     ) -> TransactionFut {
-        self.platform_coin.send_maker_spends_taker_payment(
-            taker_payment_tx,
-            time_lock,
-            taker_pub,
-            secret,
-            secret_hash,
-            swap_contract_address,
-            swap_unique_data,
-        )
+        self.platform_coin
+            .send_maker_spends_taker_payment(maker_spends_payment_args.clone())
```

Comment:
We don't need to clone the args here and in other methods

Reply:
done